modules/core/src/test/java/org/apache/ignite/internal/processors/cache/datastructures/GridCacheSetAbstractSelfTest.java

```diff
@@ -1014,28 +1026,56 @@ public void testCacheReuse() throws Exception {
 
         colCfg.setAtomicityMode(ATOMIC);
         colCfg.setGroupName("grp1");
+        colCfg.setCollocated(true);
 
         IgniteSet set1 = ignite.set("set1", colCfg);
         IgniteSet set2 = ignite.set("set2", colCfg);
 
         assert cctx(set1).cacheId() == cctx(set2).cacheId();
 
-        colCfg.setAtomicityMode(TRANSACTIONAL);
+        colCfg.setCollocated(false);
 
         IgniteSet set3 = ignite.set("set3", colCfg);
         IgniteSet set4 = ignite.set("set4", colCfg);
 
-        assert cctx(set3).cacheId() == cctx(set4).cacheId();
-        assert cctx(set1).cacheId() != cctx(set3).cacheId();
+        assert cctx(set3).cacheId() != cctx(set4).cacheId();
         assert cctx(set1).groupId() == cctx(set3).groupId();
+        assert cctx(set3).groupId() == cctx(set4).groupId();
 
-        colCfg.setGroupName("gtp2");
+        colCfg.setAtomicityMode(TRANSACTIONAL);
+        colCfg.setCollocated(true);
 
         IgniteSet set5 = ignite.set("set5", colCfg);
         IgniteSet set6 = ignite.set("set6", colCfg);
 
         assert cctx(set5).cacheId() == cctx(set6).cacheId();
-        assert cctx(set1).groupId() != cctx(set5).groupId();
+        assert cctx(set1).cacheId() != cctx(set5).cacheId();
```

Comment:
discussion - should we replace with junit?
With what junit assertEquals or assertTrue